pymbolic/geometric_algebra/__init__.py

```diff
@@ -533,7 +542,10 @@ class MultiVector(Generic[CoeffT]):
 
     def __init__(
                 self,
-                data: Mapping[tuple[int, ...] | int, CoeffT] | np.ndarray | CoeffT,
+                data: (Mapping[int, CoeffT]
+                       | Mapping[tuple[int, ...], CoeffT]
+                       | np.ndarray
+                       | CoeffT),
```

Comment:
As far as I understand, `Mapping` is invariant in its keys and for `tuple[int, ...] | int` to work as expected, it would have to be covariant?

Reply:
Yes, good catch.